packages/wallet/src/redux/protocols/virtual-funding/reducer.ts

```diff
@@ -301,49 +301,80 @@ function waitForGuarantorFundingReducer(
   action: WalletAction,
 ) {
   const { processId, protocolLocator } = protocolState;
-  if (routesToIndirectFunding(action, protocolLocator)) {
-    const result = indirectFunding.indirectFundingReducer(
-      protocolState.indirectGuarantorFunding,
+  if (
```

Comment:
Maybe this function could use a local enum like (ignore naming)
```
enum Direction { IndirectFunding, ConsensusUpdate, Other }
const direction: Direction =
  routesToIndirectFunding(action, protocolLocator) ? Direction.IndirectFunding :
  routesToConsensusUpdate(action, protocolLocator) ? Direction.ConsensusUpdate :
  Direction.Other

switch (direction) {
  ...
}
```

Reply:
It looks like we don't get the type narrowing if we try to do it this way unfortunately.

Reply:
Good point. At the least, it would be good to use `if (expr1) { ... } else if (expr2) { ... }`, which is semantically equivalent?